packages/template/vite-typescript/tmpl/vite.renderer.config.ts

```diff
@@ -0,0 +1,4 @@
+import { defineConfig } from 'vite';
+
+// https://vitejs.dev/config
+export default defineConfig({});
```

Comment:
Looks like Vite's default cache directory is `node_modules/.vite`, so we end up packaging that folder along with the project dependencies when building the project. Do you think it would be possible to use `cacheDir: './.vite'` as the default in the Vite configs, either here  or maybe directly in the Vite plugin?

Reply:
The directory `node_modules/.vite` is `esbuild` [Pre-Bundling](https://vitejs.dev/guide/dep-pre-bundling.html#dependency-pre-bundling) cache dir, it's only work `vite serve` phase and ignored `vite build` phase by default.
Even if it is used in the `vite build` phase, it is rebuild by Vite into the `dist` folder.

Reply:
> The directory node_modules/.vite is esbuild [Pre-Bundling](https://vitejs.dev/guide/dep-pre-bundling.html#dependency-pre-bundling) cache dir

That's why we don't need it in `node_modules` when the app is packaged - the contents of this folder are not necessary for the app to run, but it's being included anyway 😅

In my case, I tested the template with a hello world React app (just `electron-squirrel-startup`, `react` and `react-dom` as dependencies), and my `out/<project_name>/resources/app/node_modules/.vite` folder was about 2.5 MB. No big deal, but I'm thinking it might become more of a problem for larger apps with multiple dependencies, so it might be worth addressing that somehow (maybe just deleting the folder from `node_modules` before packaging or using the `packageAfterCopy` hook could do the trick here).

That being said, I have zero experience with Vite and I don't how large this folder can get - if it never exceeds a couple megabytes, it might not be worth the hassle.

Reply:
I think we can discuss this further in another Issue / PR. If it’s really necessary and the implementation is not complicated, we can add a “Good first issue” tag to this particular issue.

> This PR has been open for 5 months now, and if there are no obvious issues, my personal suggestion would be to proceed with the merge.

PTAL @erickzhao @dsanders11

Reply:
Sounds good to me 👍

Reply:
Sounds good to me too. Let's fix both at the same time.